community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/logical/PlanSingleQuery.scala

```diff
@@ -62,18 +62,17 @@ case class PlanSingleQuery(planPart: (PlannerQuery, LogicalPlanningContext, Opti
       val leaves = plan.leaves.collect {
         case n: NodeLogicalLeafPlan => n.idName
       }
-
-
       //if we have unsafe rels we need to check relation overlap and delete
       //overlap immediately
       (hasUnsafeRelationships(plannerQuery) &&
-        (relationshipOverlap(plannerQuery) || deleteOverlap(plannerQuery))) ||
+        (plannerQuery.updateGraph.relationshipOverlap(plannerQuery.queryGraph) ||
```

Comment:
I changed my mind on those, but apparently you didn't :) I'm fine with this though.